flink-htm-streaming-java/src/main/java/org/numenta/nupic/flink/serialization/KryoSerializer.java

```diff
@@ -123,17 +139,74 @@ public InputStream toInputStream() {
     }
 
     /**
+     * The HTM serializer handles the Persistable callbacks automatically, but
+     * this method is for any additional actions to be taken.
+     * @param t the instance to be serialized.
+     */
+    protected void preSerialize(T t) {
+    }
+
+    /**
+     * The HTM serializer handles the Persistable callbacks automatically, but
+     * this method is for any additional actions to be taken.
+     * @param t the instance newly deserialized.
+     */
+    protected void postDeSerialize(T t) {
+    }
+
+    /**
      * Register the HTM types with the Kryo serializer.
      * @param config
      */
     public static void registerTypes(ExecutionConfig config) {
         for(Class<?> c : SerialConfig.DEFAULT_REGISTERED_TYPES) {
-            config.registerTypeWithKryoSerializer(c, (Class<? extends Serializer<?>>) (Class<?>) KryoSerializer.class);
+            Class<?> serializerClass = DEFAULT_SERIALIZERS.getOrDefault(c, (Class<?>) KryoSerializer.class);
+            config.registerTypeWithKryoSerializer(c, (Class<? extends Serializer<?>>) serializerClass);
         }
         for(Class<?> c : KryoSerializer.ADDITIONAL_REGISTERED_TYPES) {
-            config.registerTypeWithKryoSerializer(c, (Class<? extends Serializer<?>>) (Class<?>) KryoSerializer.class);
+            Class<?> serializerClass = DEFAULT_SERIALIZERS.getOrDefault(c, (Class<?>) KryoSerializer.class);
+            config.registerTypeWithKryoSerializer(c, (Class<? extends Serializer<?>>) serializerClass);
         }
     }
 
     static final Class<?>[] ADDITIONAL_REGISTERED_TYPES = { Network.class };
```

Comment:
As far as I can tell there is no reason to register the Network object. Internally, all registering does is share the class name (string) among all instances of that particular class (which is slower). So pre-registering helps avoid this behavior, but since there is only ever one instance of the Network, I don't see how registering it will be any more efficient because there is only ever one of its type being serialized?